packages/application-extension/src/index.ts

```diff
@@ -200,16 +200,24 @@ const opener: JupyterFrontEndPlugin<void> = {
   ): void => {
     const { commands, docRegistry } = app;
 
+    const ignoredPattern = new RegExp('/tree/(.*)');
+    const pathSegmentPattern = new RegExp('\\bnotebooks\\b|\\bedit\\b');
     const command = 'router:tree';
     commands.addCommand(command, {
       execute: (args: any) => {
         const parsed = args as IRouter.ILocation;
         const matches = parsed.path.match(TREE_PATTERN) ?? [];
+
         const [, , path] = matches;
         if (!path) {
           return;
         }
 
+        const pathSegments: string[] = parsed.path.split(pathSegmentPattern);
+        if (pathSegments.length > 1 && pathSegments[0].match(ignoredPattern)) {
```

Comment:
Wondering if following a similar logic as in JupyterLab (disposing the command after routing) could help keep things simpler and avoid special cases using patterns? 

https://github.com/jupyterlab/jupyterlab/blob/cdca0a20ba84af21922c390c46e086e50120d4cc/packages/application-extension/src/index.tsx#L938-L939